src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/service/UserService.kt

```diff
@@ -545,8 +548,34 @@ class UserService(
     )
   }
 
-  private fun toUserDetail(user: UserPersonDetail) =
-    UserDetail(user, accountDetailRepository.findById(user.username).orElse(AccountDetail(username = user.username)))
+  private fun toUserDetail(user: UserPersonDetail): UserDetail {
+    val accountDetail = accountDetailRepository.findById(user.username).orElse(AccountDetail(username = user.username))
+    return UserDetail(
+      user,
+      accountDetail,
+      isAccountNonLocked(accountDetail.status),
+      isCredentialsNonExpired(accountDetail),
+      isEnabled(user, accountDetail.status)
+    )
+  }
+
+  fun isAccountNonLocked(accountStatus: AccountStatus): Boolean =
+    EnumSet.of(AccountStatus.OPEN, AccountStatus.EXPIRED, AccountStatus.EXPIRED_GRACE)
+      .contains(accountStatus)
+
+  fun isCredentialsNonExpired(accountDetail: AccountDetail): Boolean {
+    val statusNonExpired =
+      !EnumSet.of(AccountStatus.EXPIRED, AccountStatus.EXPIRED_LOCKED, AccountStatus.EXPIRED_LOCKED_TIMED).contains(
+        accountDetail.status
+      )
+    val passwordExpiry = accountDetail.passwordExpiry
+    return statusNonExpired && (passwordExpiry == null || passwordExpiry.isAfter(LocalDateTime.now()))
+  }
+
+  fun isEnabled(user: UserPersonDetail, accountStatus: AccountStatus): Boolean {
+    return user.staff.isActive && EnumSet.of(AccountStatus.OPEN, AccountStatus.EXPIRED, AccountStatus.EXPIRED_GRACE)
```

Comment:
This is a combination of `isActive` and the logic from [hmpps-auth/NomisUserPersonDetails](https://github.com/ministryofjustice/hmpps-auth/blob/a5e82abdefee74b3ec230c86013c6ad1c46a3634/src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/nomis/model/NomisUserPersonDetails.kt#L123-L126)

Reply:
Hence dropping `isActive` from `UserDetail`